components/dashboard/src/workspaces/ConnectToSSHModal.tsx

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { useState } from "react";
+import Modal from "../components/Modal";
+import Tooltip from "../components/Tooltip";
+import copy from "../images/copy.svg";
+import CodeText from "../components/CodeText";
+
+function InputWithCopy(props: { value: string; tip?: string; className?: string }) {
+    const [copied, setCopied] = useState<boolean>(false);
+    const copyToClipboard = (text: string) => {
+        const el = document.createElement("textarea");
+        el.value = text;
+        document.body.appendChild(el);
+        el.select();
+        try {
+            document.execCommand("copy");
+        } finally {
+            document.body.removeChild(el);
+        }
+        setCopied(true);
+        setTimeout(() => setCopied(false), 2000);
+    };
+    const tip = props.tip ?? "Click to copy";
+    return (
+        <div className={`w-full relative ${props.className ?? ""}`}>
+            <input
+                disabled={true}
+                readOnly={true}
+                autoFocus
+                className="w-full pr-8 overscroll-none"
+                type="text"
+                defaultValue={props.value}
+            />
+            <div className="cursor-pointer" onClick={() => copyToClipboard(props.value)}>
+                <div className="absolute top-1/3 right-3">
+                    <Tooltip content={copied ? "Copied!" : tip}>
+                        <img src={copy} alt="copy icon" title={tip} />
+                    </Tooltip>
+                </div>
+            </div>
+        </div>
+    );
+}
+
+interface SSHProps {
+    workspaceId: string;
+    ownerToken: string;
+    ideUrl: string;
+}
+
+function SSHView(props: SSHProps) {
+    const sshCommand = `ssh ${props.workspaceId}#${props.ownerToken}@${props.ideUrl}`;
+    return (
+        <div className="border-t border-b border-gray-200 dark:border-gray-800 mt-2 -mx-6 px-6 py-6">
+            <div className="mt-1 mb-4">
+                <p className="text-gray-500 whitespace-normal text-base">
+                    The following shell command can be used to SSH into this workspace.
+                </p>
+                <p className="mt-2 text-gray-500 whitespace-normal text-base">
+                    If you encounter a prompt to enter a password, please make sure you have a private key in&nbsp;
+                    <CodeText>~/.ssh/id_rsa</CodeText>. You can use command&nbsp;<CodeText>ssh-keygen</CodeText>&nbsp;to
+                    generate one.
+                </p>
```

Comment:
**suggestion**: What do you think of using an alert component here for sufacing this information and rephrasing the copy to avoid any confusion because of the different options available (different SSH key types) as well as different set ups (where different systems store SSH keys).

| BEFORE | AFTER |
|-|-|
| <img width="1440" alt="Screenshot 2022-04-05 at 12 40 19 PM" src="https://user-images.githubusercontent.com/120486/161741493-561603d1-cc8f-48b1-892d-74d74e510a47.png"> | <img width="1440" alt="Screenshot 2022-04-05 at 1 04 48 PM" src="https://user-images.githubusercontent.com/120486/161741502-f0371b78-6480-4266-9412-e8341168b959.png"> |

**fyi**: The alert component is being refactored in https://github.com/gitpod-io/gitpod/pull/8783 by @mustard-mh, but it should be fine to use the existing info alert component variant. Re-posting from https://github.com/gitpod-io/gitpod/pull/8783#discussion_r833317085 for visibility:

> [FWIW](https://en.wiktionary.org/wiki/FWIW), we also have https://github.com/gitpod-io/gitpod/issues/7613 to tackle this component separately and also a few similar components that we could deprecate in favor of this, see [AlertBox](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/src/components/AlertBox.tsx) and [InfoBox](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/src/components/InfoBox.tsx). However deprecating those other component could be tackled separately.

Reply:
**suggestion**: Following the comment in https://github.com/gitpod-io/gitpod/pull/9003#issuecomment-1085719584, if we'd like to include the security warning about access, we could restructure the modal like the following:

| Proposed Modal |
|-|
| <img width="1440" alt="Screenshot 2022-04-05 at 1 38 51 PM" src="https://user-images.githubusercontent.com/120486/161745900-068255ec-0c1f-414d-a578-4ed5ce049d8c.png"> |

I've also included a sentence about the temporary generated token used in the command. Let me know what you think. We could also move that sentence out of the alerts. 💭

Reply:
Marking this one as resolved, too.